mqtt/mqtt-bridge/src/persist/storage/error.rs

```diff
@@ -0,0 +1,35 @@
+use crate::persist::storage::ring_buffer::error::RingBufferError;
+
+use bincode::Error as BincodeError;
+use sled::Error as SledError;
+use thiserror::Error;
+
+// create_error!(StorageError);
+
+#[derive(Debug, Error)]
+pub enum StorageError {
+    #[error("RingBuffer error occurred")]
+    RingBuffer(#[from] RingBufferError),
+    #[error("Database error occurred")]
+    Database(#[from] SledError),
+    #[error("Serialization error occurred")]
+    Serialization(#[from] BincodeError),
+}
+
+// impl From<RingBufferError> for StorageError {
```

Comment:
Was trying different ways to create Errors, and seems we use the thiserror crate. Will remove these comments, but needed them while I was figuring out thiserror.